git-branchless-smartlog/src/lib.rs

```diff
@@ -851,8 +857,13 @@ pub fn smartlog(
             &mut DifferentialRevisionDescriptor::new(&repo, &Redactor::Disabled)?,
             &mut CommitMessageDescriptor::new(&Redactor::Disabled)?,
         ],
-    )?;
-    for line in lines {
+    )?
+    .into_iter();
+    while let Some(line) = if reverse {
```

Comment:
Using an `if` in the iteration expression here is a bit clunky, but all the alternatives I could think of added more code than seemed necessary. (E.g.: splitting out a separate function, adding a local `&mut dyn Iterator<...>` variable backed by one of two conditionally initialized iterator types, ...)

Reply:
A classic Rust ergonomics problem! Looks fine to me.